src/rgw/rgw_sal_motr.cc

```diff
@@ -1933,8 +1933,20 @@ int MotrObject::MotrReadOp::prepare(optional_yield y, const DoutPrefixProvider*
   if (rc < 0)
     return rc;
 
-  if (ent.is_delete_marker())
+  req_state* s = static_cast<req_state*>(rctx->get_private());
+
+  // In GET/HEAD object API, return "MethodNotAllowed"
+  // if delete-marker is the latest object entry
+  // Else, return "NoSuchKey" error
+  if (ent.is_delete_marker()) {
+    if (source->get_instance() == ent.key.instance && ent.key.instance != "") {
+      ldpp_dout(dpp, LOG_DEBUG) <<__func__ << "The GET/HEAD object with version-id of "
+                                            "delete-marker is not allowed." << dendl;
```

Comment:
```suggestion
      ldpp_dout(dpp, LOG_DEBUG) <<__func__ << ": GET/HEAD object with version-id of "
                                            "delete-marker is not allowed." << dendl;
```

Reply:
Yeah, corrected the statement now. Thanks!
Also, added DEBUG word in log statement
```
ldpp_dout(dpp, LOG_DEBUG) <<__func__ << ": DEBUG: The GET/HEAD object with version-id of "
                                            "delete-marker is not allowed." << dendl;
```